client-csharp/SessionPool.cs

```diff
@@ -0,0 +1,642 @@
+            return util_functions.verify_success(status, SUCCESS_CODE);
+        }
+
+        public async Task<int> create_time_series_async(string ts_path, TSDataType data_type, TSEncoding encoding, Compressor compressor){
+            TSStatus status;
+            var client = client_lst.Take();
+            var req = new TSCreateTimeseriesReq(client.sessionId, ts_path, (int)data_type, (int)encoding, (int)compressor);
+            try{
+                status = await client.client.createTimeseriesAsync(req);
+            }
+            catch(TException e){
+                client_lst.Add(client); 
+                var err_msg = String.Format("create time series {0} failed", ts_path);   
+                throw new TException(err_msg, e);
+            }
+            if(debug_mode){
+                _logger.Info("creating time series {0} successfully, server message is {1}", ts_path, status.Message);
+            }
+            client_lst.Add(client); 
+
+            return util_functions.verify_success(status, SUCCESS_CODE); 
+        }
+        public async Task<int> delete_storage_group_async(string group_name){
+            TSStatus status;
+            var client = client_lst.Take();
+            try{
+                status = await client.client.deleteStorageGroupsAsync(client.sessionId, new List<string>{group_name});
+            }
+            catch(TException e){
+                client_lst.Add(client);
+                var err_msg = String.Format("delete storage group {0} failed", group_name);
+                throw new TException(err_msg, e);
+            }
+            if(debug_mode){
+                var message = String.Format("delete storage group {0} successfully, server message is {1}", group_name, status.Message);
+                _logger.Info(message);
+            }
+            client_lst.Add(client);
+            return util_functions.verify_success(status, SUCCESS_CODE);
+        }
+        public async Task<int> delete_storage_groups_async(List<string> group_names){
+            var client = client_lst.Take();
+
+            TSStatus status;
+            try{
+                status = await client.client.deleteStorageGroupsAsync(client.sessionId, group_names);
+                
+            }
+            catch(TException e){
+                client_lst.Add(client);
+                var err_msg = String.Format("delete storage group(s) {0} failed", group_names);
+                throw new TException(err_msg, e);           
+            }
+            if(debug_mode){
+                _logger.Info("delete storage group(s) {0} successfully, server message is {1}", group_names, status.Message);
+            }
+            client_lst.Add(client);
+            return util_functions.verify_success(status, SUCCESS_CODE);
+        }
+        public async Task<int> create_multi_time_series_async(List<string> ts_path_lst, List<TSDataType> data_type_lst, List<TSEncoding> encoding_lst, List<Compressor> compressor_lst){
```

Comment:
Based on the Microsoft naming [conventions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions#pascal-case) all methods should be in Pascal Case

Reply:
Yes, great idea, this will be more friendly to csharp users